rustls/src/msgs/fragmenter.rs

```diff
@@ -26,41 +23,30 @@ impl MessageFragmenter {
 
     /// Take the Message `msg` and re-fragment it into new
     /// messages whose fragment is no more than max_frag.
-    /// The new messages are appended to the `out` deque.
-    /// Payloads are copied.
-    pub fn fragment(&self, msg: PlainMessage, out: &mut VecDeque<PlainMessage>) {
-        // Non-fragment path
-        if msg.payload.0.len() <= self.max_frag {
-            out.push_back(msg);
-            return;
-        }
-
-        for chunk in msg.payload.0.chunks(self.max_frag) {
-            out.push_back(PlainMessage {
-                typ: msg.typ,
-                version: msg.version,
-                payload: Payload(chunk.to_vec()),
-            });
-        }
+    /// Return an iterator across those messages.
+    /// Payloads are borrowed.
+    pub fn fragment_message<'a>(
```

Comment:
Since this is now a one-line wrapper of `fragment_slice()` used only in tests and for fuzzing, can we just get rid of this? We can potentially redo it as a helper function inside `tests`?

Reply:
I'd be happy to consolidate on just one of these two, but FWIW it's not just tests and fuzzing.

fragment_message is used in send_msg and send_msg_encrypt in conn.rs.

fragment_slice is used in send_appdata_encrypt in conn.rs.

Since fragment_slice is more generally applicable (fragment_message requires the caller to have an owned PlainMessage), I'll consolidate on that one (and rename to `fragment` as you suggest).

Reply:
Ah, sorry, I missed that. In that case I think your previous set up was better, let's ditch the commit you added for it?